fetch.js

```diff
@@ -463,4 +489,6 @@
     })
   }
   self.fetch.polyfill = true
-})(typeof self !== 'undefined' ? self : this);
```

Comment:
You've removed this construct which was here to make `fetch` work for both normal browser and [webworker](https://github.com/github/fetch/blob/27473e937c6bdbb61212cc078ad60d22ae77ffb1/test/test-worker.html) environments. You've replaced it with two different contexts: `this` for the concept of the "root" in the UMD shim, and `window
 || global` for the concept of "main" inside the factory function. This is likely what broke the test suite for webworker.

I think the concepts of "root" and "main" should be unified and should be passed around as a single reference. Also, per https://github.com/github/fetch/pull/521, I think the fallback chain for getting a reference to the global object should be (pseudocode) `self || this || global`.